### PR TITLE
feature: S3C-3568 Convert Existing Marshalling/ Unmarshalling to Generic 

### PR DESCRIPTION
**Pull request by saitharunthodupunuru**
1. Moved the existing basic and common code (like initiating, invoking Vault API) to BaseServicesClient.
So AccountServicesClient will hold only business logic for Account related web Services
1. Converted the existing Marshalling/Unmarshalling classes to Generic Marshalling/Unmarshalling classes to ease future engineering. All the future developments can just extend the generic classes with ease.

Note to Reviewers:
* Please go through each and every commit to get detailed understanding of the changes

### CODE REVIEW

File: pmd-rules.xml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    <rule ref="category/java/bestpractices.xml">
+        <exclude name="JUnitAssertionsShouldIncludeMessage"/>
+        <exclude name="JUnitTestContainsTooManyAsserts"/>
+        <exclude name="JUnit4TestShouldUseAfterAnnotation"/>
+        <exclude name="JUnit4TestShouldUseBeforeAnnotation"/>
+        <exclude name="UseVarargs"/>
+        <exclude name="AccessorMethodGeneration"/>
+        <exclude name="UnusedFormalParameter"/>
+    </rule>
+</ruleset>
```
Comment (anurag4DSB) [inline pmd-rules.xml:32]: missing new line

File: vaultclient/src/main/java/com/scality/vaultclient/services/BaseServicesClient.java [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+        this(DefaultAWSCredentialsProviderChain.getInstance(), clientConfiguration);
+    }
+
+    /*public AccountServicesClient(ClientConfiguration clientConfiguration, RequestMetricCollector requestMetricCollector) {
+        super(clientConfiguration, requestMetricCollector);
+    }
+
+    protected AccountServicesClient(ClientConfiguration clientConfiguration, RequestMetricCollector requestMetricCollector, boolean disableStrictHostNameVerification) {
+        super(clientConfiguration, requestMetricCollector, disableStrictHostNameVerification);
+    }*/
```
Comment (anurag4DSB) [inline vaultclient/src/main/java/com/scality/vaultclient/services/BaseServicesClient.java:82]: Do we need this?

File: vaultclient/src/main/java/com/scality/vaultclient/services/BaseServicesClient.java [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+     * ExecutionContext beforehand.
+     **/
+    private <X, Y extends AmazonWebServiceRequest> Response<X> doInvoke(Request<Y> request, HttpResponseHandler<AmazonWebServiceResponse<X>> responseHandler,
+                                                                        ExecutionContext executionContext, URI discoveredEndpoint, URI uriFromEndpointTrait) {
+        request.setEndpoint(endpoint);
+        request.setTimeOffset(timeOffset);
+
+        DefaultErrorResponseHandler errorResponseHandler = new DefaultErrorResponseHandler(getExceptionUnmarshallers());
+
+//        return client.execute(request, responseHandler, errorResponseHandler, executionContext);
```
Comment (anurag4DSB) [inline vaultclient/src/main/java/com/scality/vaultclient/services/BaseServicesClient.java:314]: may be a leftover?

**Review by anurag4DSB — COMMENTED** [review]
Some leftover commented code, otherwise LGTM

File: pmd-rules.xml (hunk 1 above)
Comment (saitharunthodupunuru) [reply pmd-rules.xml]: added in ad6fa35

File: vaultclient/src/main/java/com/scality/vaultclient/services/BaseServicesClient.java (hunk 2 above)
Comment (saitharunthodupunuru) [reply vaultclient/src/main/java/com/scality/vaultclient/services/BaseServicesClient.java]: fixed in ad6fa35

File: vaultclient/src/main/java/com/scality/vaultclient/services/BaseServicesClient.java (hunk 3 above)
Comment (saitharunthodupunuru) [reply vaultclient/src/main/java/com/scality/vaultclient/services/BaseServicesClient.java]: fixed in ad6fa35

**Review by anurag4DSB — APPROVED** [review]

**Review by ilkescality — APPROVED** [review]

**Pull request merged**